src/transformers/models/wav2vec2/modeling_flax_wav2vec2.py

```diff
@@ -0,0 +1,482 @@
+        self.activation = ACT2FN[self.config.feat_extract_activation]
+
+    def __call__(self, hidden_states):
+        hidden_states = self.conv(hidden_states)
+
+        hidden_states = hidden_states.transpose(-2, -1)
+        hidden_states = self.layer_norm(hidden_states)
+        hidden_states = hidden_states.transpose(-2, -1)
+
+        hidden_states = self.activation(hidden_states)
+        return hidden_states
+
+
+class FlaxWav2Vec2GroupNormConvLayer(nn.Module):
+    config: Wav2Vec2Config
+    layer_id: int = 0
+    dtype: jnp.dtype = jnp.float32
+
+    def setup(self):
+        self.in_conv_dim = self.config.conv_dim[self.layer_id] if self.layer_id > 0 else 1
+        self.out_conv_dim = self.config.conv_dim[self.layer_id]
+
+        self.conv = nn.Conv(
+            features=self.out_conv_dim,
+            kernel_size=self.config.conv_kernel[self.layer_id],
+            use_bias=self.config.conv_bias,
+            kernel_init=jax.nn.initializers(self.config.initializer_range, self.dtype),
+            dtype=self.dtype,
+        )
+        self.layer_norm = nn.GroupNorm(
+            num_groups=self.out_conv_dim, group_size=self.out_conv_dim, epsilon=1e-5, dtype=self.dtype
+        )
+        self.activation = ACT2FN[self.config.feat_extract_activation]
+
+    def __call__(self, hidden_states):
+        hidden_states = self.conv(hidden_states)
+        hidden_states = self.layer_norm(hidden_states)
+        hidden_states = self.activation(hidden_states)
+        return hidden_states
+
+
+class FlaxWav2Vec2PositionalConvEmbedding(nn.Module):
+    config: Wav2Vec2Config
+    dtype: jnp.dtype = jnp.float32
+
+    def setup(self):
+        self.conv = nn.Conv(
+            features=self.config.hidden_size,
+            kernel_size=self.config.num_conv_pos_embeddings,
+            kernel_init=jax.nn.initializers(self.config.initializer_range, self.dtype),
+            padding=(self.config.num_conv_pos_embeddings // 2,),
+            feature_group_count=self.config.num_conv_pos_embeddings,
+            dtype=self.dtype,
+        )  # TODO (PS): weigh_norm ?
+        self.activation = ACT2FN[self.config.feat_extract_activation]
+        self.num_pad_remove = 1 if self.config.num_conv_pos_embeddings % 2 == 0 else 0
+
+    def __call__(self, hidden_states):
+        hidden_states = hidden_states.transpose(1, 2)
+
+        hidden_states = self.conv(hidden_states)
+        if self.num_pad_remove > 0:
+            hidden_states = hidden_states[:, :, : -self.num_pad_remove]
+        hidden_states = self.activation(hidden_states)
+
+        hidden_states = hidden_states.transpose(1, 2)
+        return hidden_states
+
+
+class FlaxWav2Vec2FeatureExtractor(nn.Module):
+    """Construct the featurs from raw audio waveform"""
+
+    config: Wav2Vec2Config
+    dtype: jnp.dtype = jnp.float32
+
+    def setup(self):
+        config = self.config
+        if config.feat_extract_norm == "group":
```

Comment:
For now, can we only allow `elif config.feat_extract_norm == "layer"`? 

Here we can just raise an Error instead:

```python
raise NotImplementedError("At the moment flax wav2vec2 only supports ``config.feat_extact_norm == 'layer'``")
```